File: ClassEval_T/cpp/solution/SignInSystem.cpp
#include <unordered_map>
#include <vector>
#include <string>

class SignInSystem {
public:
    SignInSystem();
    bool add_user(const std::string& username);
    bool sign_in(const std::string& username);
    bool check_sign_in(const std::string& username);
    bool all_signed_in();
    std::vector<std::string> all_not_signed_in();

private:
    std::unordered_map<std::string, bool> users;
};

SignInSystem::SignInSystem() {}

bool SignInSystem::add_user(const std::string& username) {
    auto result = users.find(username);
    if (result != users.end()) {
        return false;
    }
    else {
        users[username] = false;
        return true;
    }
}

bool SignInSystem::sign_in(const std::string& username) {
    auto result = users.find(username);
    if (result == users.end()) {
        return false;
    }
    else {
        users[username] = true;
        return true;
    }
}

bool SignInSystem::check_sign_in(const std::string& username) {
    auto result = users.find(username);
    if (result == users.end()) {
        return false;
    }
    else {
        return result->second;
    }
}

bool SignInSystem::all_signed_in() {
    for (const auto& pair : users) {
        if (!pair.second) {
            return false;
        }
    }
    return true;
}

std::vector<std::string> SignInSystem::all_not_signed_in() {
    std::vector<std::string> not_signed_in_users;
    for (const auto& pair : users) {
        if (!pair.second) {
            not_signed_in_users.push_back(pair.first);
        }
    }
    return not_signed_in_users;
}

File: ClassEval_T/java/solutuon/SignInSystem.java
package org.example;

import java.util.HashMap;
import java.util.Map;
import java.util.ArrayList;
import java.util.List;

public class SignInSystem {
    private Map<String, Boolean> users;

    public SignInSystem() {
        this.users = new HashMap<>();
    }

    public boolean addUser(String username) {
        if (users.containsKey(username)) {
            return false;
        } else {
            users.put(username, false);
            return true;
        }
    }

    public boolean signIn(String username) {
        if (!users.containsKey(username)) {
            return false;
        } else {
            users.put(username, true);
            return true;
        }
    }

    public boolean checkSignIn(String username) {
        if (!users.containsKey(username)) {
            return false;
        } else {
            return users.get(username);
        }
    }

    public boolean allSignedIn() {
        for (boolean signedIn : users.values()) {
            if (!signedIn) {
                return false;
            }
        }
        return true;
    }

    public List<String> allNotSignedIn() {
        List<String> notSignedInUsers = new ArrayList<>();
        for (Map.Entry<String, Boolean> entry : users.entrySet()) {
            if (!entry.getValue()) {
                notSignedInUsers.add(entry.getKey());
            }
        }
        return notSignedInUsers;
    }
}